From d1719aebf357248ed20614462a5a614920ac2810 Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Tue, 28 Jul 2020 22:09:31 -0700
Subject: [PATCH 1471/1852] staging: xlnxsync: Don't write back Interrupt
 status register(ISR)

commit 8582c969b020980b90540a15090549403f6a1938 from
https://github.com/Xilinx/linux-xlnx.git

SyncIP ISR register is Read to Clear, no need to write it back after read.

Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xlnxsync/xlnxsync.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/staging/xlnxsync/xlnxsync.c b/drivers/staging/xlnxsync/xlnxsync.c
index bcd6b3b28e0e..cec913334a37 100644
--- a/drivers/staging/xlnxsync/xlnxsync.c
+++ b/drivers/staging/xlnxsync/xlnxsync.c
@@ -966,7 +966,6 @@ static irqreturn_t xlnxsync_irq_handler(int irq, void *data)
 		u32 i, j;
 
 		val = xlnxsync_read(xlnxsync, chan->id, XLNXSYNC_ISR_REG);
-		xlnxsync_write(xlnxsync, chan->id, XLNXSYNC_ISR_REG, val);
 
 		if (val & XLNXSYNC_ISR_SYNC_FAIL_MASK)
 			chan->sync_err = true;
-- 
2.31.1

